File: push/src/main/java/com/xiaomi/push/service/timers/HybridTimer.java
package com.xiaomi.push.service.timers;

import android.content.Context;
import com.xiaomi.channel.commonutils.misc.DateTimeHelper;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/service/timers/HybridTimer.class */
class HybridTimer extends AlarmManagerTimer {
    private static int pingInterval = DateTimeHelper.HOUR_IN_MS;

    public HybridTimer(Context context) {
        super(context);
    }

    @Override // com.xiaomi.push.service.timers.AlarmManagerTimer
    public long getPingInteval() {
        return pingInterval;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/service/timers/HybridTimer.kt
package com.xiaomi.push.service.timers

import android.content.Context
import com.xiaomi.channel.commonutils.misc.DateTimeHelper

/**
 * HybridTimer implements Alarm.IAlarm directly since AlarmManagerTimer
 * is in the push module which is not a dependency of legacy-runtime.
 */
class HybridTimer(context: Context) : Alarm.IAlarm {
    private val delegate = AlarmV21(context)

    override fun isAlive(): Boolean = delegate.isAlive()

    override fun registerPing(force: Boolean) = delegate.registerPing(force)

    override fun stop() = delegate.stop()

    fun getPingInteval(): Long = pingInterval

    companion object {
        private var pingInterval: Long = DateTimeHelper.HOUR_IN_MS.toLong()
    }
}
